docs/troubleshooting.md

```diff
@@ -28,14 +34,159 @@ Please clear your Azure profile directory with `rm -rf ~/.azure` and retry after
 ```
 
-__Resolution__:
+__Troubleshooting__:
 
 This is typically encountered when you attempt to install BKPR with a DNS zone (`--dns-zone`) that was used in a earlier installation on BKPR. Login to the [Azure Portal](https://portal.azure.com) and navigate to __Azure Active Directory > App registrations__ and filter the result with the keyword `kubeprod`. From the filtered results remove the entries that have the BKPR DNS zone in its name and retry the BKPR installation.
 
 ![Azure SP Conflict](images/azure-sp-conflict.png)
+
+## Troublehooting DNS
+
+### Unable to resolve DNS addresses
+
+You have installed BKPR to your Kubernetes cluster, but are unable to access any of the ingress endpoints due to DNS resolution errors.
+
+```bash
+ping prometheus.my-domain.com
+ping: prometheus.my-domain.com: Name or service not known
+```
+
+__Troubleshooting__:
+
+DNS address resolution could be a result of configuration issues. For a working DNS setup you need to ensure all of the following conditions are met.
+
+- [ExternalDNS Pods are running](#externaldns-pods-are-not-starting)
+- [ExternalDNS is updating DNS zone records](#externaldns-is-not-updating-dns-zone-records)
+- [DNS glue records are configured](#dns-glue-records-are-not-configured)
+- [DNS propagation has completed](#dns-propagation-has-not-completed)
+
+### ExternalDNS Pods are not starting
+
+Use the following command to check the status of the `external-dns` deployment:
+
+```bash
+kubectl -n kubeprod get deployments external-dns
+NAME           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+external-dns   1         1         1            0           5m
+```
+
+The `AVAILABLE` column indicates the number of Pods that have started successfully.
+
+__Troubleshooting__:
+
+A Pod goes through various [lifecyle phases](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/) before it enters the `Running` phase. You can use the following command to watch the rollout status of the `external-dns` Deployment:
+
+```bash
+kubectl -n kubeprod rollout status deployments external-dns
+Waiting for deployment "external-dns" rollout to finish: 0 of 1 updated replicas are available...
+Waiting for deployment "external-dns" rollout to finish: 1 of 1 updated replicas are available...
+```
+
+The command will return with the message `deployment "external-dns" successfully rolled out` after all the Pods in the `external-dns` Deployment have started successfully. Please note it could take a few minutes for the `external-dns` Deployment to complete.
+
+If it takes an abnormally long time (>5m) for the `external-dns` Pods to enter the `Running` phase, it indicates that it may have encountered an error condition.
```

Comment:
too many "it"s, reword as "If it takes an abnormally long time (>5m) for the `external-dns` Pods to enter the `Running` phase, they may have encountered an error condition."